src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCISharedBuildJobQueueService.java

```diff
@@ -99,6 +103,10 @@ public LocalCISharedBuildJobQueueService(HazelcastInstance hazelcastInstance, Ex
         this.processingJobs.addLocalEntryListener(new ProcessingBuildJobItemListener());
         this.buildAgentInformation.addLocalEntryListener(new BuildAgentListener());
         this.localCIBuildQueueWebsocketService = localCIBuildQueueWebsocketService;
+
+        if (env.acceptsProfiles(Profiles.of("scheduling"))) {
```

Comment:
you could add a helper method in the ProfileService like the `isBamboo` or `isLocalVcsCi` and shorten this to `this.schedulingActive = profileService.isScheduling()`. But that is nonblocking for me